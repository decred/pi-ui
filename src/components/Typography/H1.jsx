import React from "react";
import PropTypes from "prop-types";
import styles from "./styles.css";

const H1 = ({ children, className, style }) => (
  <h1 className={`${styles.header1} ${className}`} style={style}>
    {children}
  </h1>
);

H1.propTypes = {
  children: PropTypes.string,
  className: PropTypes.string,
  style: PropTypes.object
};

H1.defaultProps = {
  className: ""
};

export default H1;
