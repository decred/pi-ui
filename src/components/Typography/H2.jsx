import React from "react";
import PropTypes from "prop-types";
import styles from "./styles.css";
import { useTruncate } from "../../hooks";
import { classNames, idPropTypeCheckForTruncatedComponents } from "../../utils";

const H2 = ({
  id,
  children,
  className,
  style,
  truncate,
  linesBeforeTruncate,
  ...props
}) => {
  useTruncate(id, truncate, linesBeforeTruncate);
  return (
    <h2
      id={id}
      className={classNames(styles.header2, className)}
      style={style}
      {...props}>
      {children}
    </h2>
  );
};

H2.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  style: PropTypes.object,
  truncate: PropTypes.bool,
  linesBeforeTruncate: PropTypes.number,
  id: idPropTypeCheckForTruncatedComponents,
};

H2.defaultProps = {
  truncate: false,
  linesBeforeTruncate: 1,
};

export default H2;
