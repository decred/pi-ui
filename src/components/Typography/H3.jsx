import React from "react";
import PropTypes from "prop-types";
import styles from "./styles.module.css";
import { useTruncate } from "../../hooks";
import { classNames, idPropTypeCheckForTruncatedComponents } from "../../utils";

const H3 = ({
  id,
  children,
  className,
  style,
  truncate,
  linesBeforeTruncate,
  ...props
}) => {
  useTruncate(id, truncate, linesBeforeTruncate);
  return (
    <h3
      id={id}
      className={classNames(styles.header3, className)}
      style={style}
      {...props}>
      {children}
    </h3>
  );
};

H3.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  style: PropTypes.object,
  truncate: PropTypes.bool,
  linesBeforeTruncate: PropTypes.number,
  id: idPropTypeCheckForTruncatedComponents,
};

H3.defaultProps = {
  truncate: false,
  linesBeforeTruncate: 1,
};

export default H3;
