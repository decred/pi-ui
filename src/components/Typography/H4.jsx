import React from "react";
import PropTypes from "prop-types";
import styles from "./styles.css";
import { classNames } from "../../utils";

const H4 = ({ children, className, style }) => (
  <h4 className={classNames(styles.header4, className)} style={style}>
    {children}
  </h4>
);

H4.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  style: PropTypes.object
};

export default H4;
