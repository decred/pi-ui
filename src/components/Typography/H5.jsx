import React from "react";
import PropTypes from "prop-types";
import styles from "./styles.css";
import { useTruncate } from "../../hooks";
import { classNames, idPropTypeCheckForTruncatedComponents } from "../../utils";

const H5 = ({
  id,
  children,
  className,
  style,
  truncate,
  linesBeforeTruncate,
  ...props
}) => {
  useTruncate(id, truncate, linesBeforeTruncate);
  return (
    <h5
      id={id}
      className={classNames(styles.header5, className)}
      style={style}
      {...props}>
      {children}
    </h5>
  );
};

H5.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  style: PropTypes.object,
  truncate: PropTypes.bool,
  linesBeforeTruncate: PropTypes.number,
  id: idPropTypeCheckForTruncatedComponents,
};

H5.defaultProps = {
  truncate: false,
  linesBeforeTruncate: 1,
};

export default H5;
