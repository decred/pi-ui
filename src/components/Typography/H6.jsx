import React from "react";
import PropTypes from "prop-types";
import styles from "./styles.module.css";
import { useTruncate } from "../../hooks";
import { classNames, idPropTypeCheckForTruncatedComponents } from "../../utils";

const H6 = ({
  id,
  children,
  className,
  style,
  truncate,
  linesBeforeTruncate
}) => {
  useTruncate(id, truncate, linesBeforeTruncate);
  return (
    <h6 id={id} className={classNames(styles.header6, className)} style={style}>
      {children}
    </h6>
  );
};

H6.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  style: PropTypes.object,
  truncate: PropTypes.bool,
  linesBeforeTruncate: PropTypes.number,
  id: idPropTypeCheckForTruncatedComponents
};

H6.defaultProps = {
  truncate: false,
  linesBeforeTruncate: 1
};

export default H6;
