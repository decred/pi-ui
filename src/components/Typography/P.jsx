import React from "react";
import PropTypes from "prop-types";
import styles from "./styles.css";
import { classNames } from "../../utils";

const P = ({ children, className, style }) => (
  <p className={classNames(styles.paragraph, className)} style={style}>
    {children}
  </p>
);

P.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  style: PropTypes.object
};

export default P;
