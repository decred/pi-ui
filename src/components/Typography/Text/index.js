export { default } from "./Text.jsx";
