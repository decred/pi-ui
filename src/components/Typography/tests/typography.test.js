import React from "react";
import { create } from "react-test-renderer";
import H1 from "../H1";
import H2 from "../H2";
import H3 from "../H3";
import H4 from "../H4";
import H5 from "../H5";
import H6 from "../H6";
import P from "../P";
import Text from "../Text";

describe("Typography Elements", () => {
  test("H1 Matches Snapshot", () => {
    const h1 = create(<H1>header1</H1>);
    expect(h1.toJSON()).toMatchSnapshot();
  });
  test("H2 Matches Snapshot", () => {
    const h2 = create(<H2>header1</H2>);
    expect(h2.toJSON()).toMatchSnapshot();
  });
  test("H3 Matches Snapshot", () => {
    const h3 = create(<H3>header1</H3>);
    expect(h3.toJSON()).toMatchSnapshot();
  });
  test("H4 Matches Snapshot", () => {
    const h4 = create(<H4>header1</H4>);
    expect(h4.toJSON()).toMatchSnapshot();
  });
  test("H5 Matches Snapshot", () => {
    const h5 = create(<H5>header1</H5>);
    expect(h5.toJSON()).toMatchSnapshot();
  });
  test("H6 Matches Snapshot", () => {
    const h6 = create(<H6>header1</H6>);
    expect(h6.toJSON()).toMatchSnapshot();
  });
  test("P Matches Snapshot", () => {
    const p = create(<P>header1</P>);
    expect(p.toJSON()).toMatchSnapshot();
  });
  test("Text Matches Snapshot", () => {
    const text = create(<Text>header1</Text>);
    expect(text.toJSON()).toMatchSnapshot();
  });
});
