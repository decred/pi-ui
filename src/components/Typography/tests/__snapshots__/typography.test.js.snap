// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Elements H1 Matches Snapshot 1`] = `
<h1
  className="header1"
>
  header1
</h1>
`;

exports[`Typography Elements H2 Matches Snapshot 1`] = `
<h2
  className="header2"
>
  header1
</h2>
`;

exports[`Typography Elements H3 Matches Snapshot 1`] = `
<h3
  className="header3"
>
  header1
</h3>
`;

exports[`Typography Elements H4 Matches Snapshot 1`] = `
<h4
  className="header4"
>
  header1
</h4>
`;

exports[`Typography Elements H5 Matches Snapshot 1`] = `
<h5
  className="header5"
>
  header1
</h5>
`;

exports[`Typography Elements H6 Matches Snapshot 1`] = `
<h6
  className="header6"
>
  header1
</h6>
`;

exports[`Typography Elements P Matches Snapshot 1`] = `
<p
  className="paragraph"
>
  header1
</p>
`;

exports[`Typography Elements Text Matches Snapshot 1`] = `
<span
  className="text textAlignStart weightRegular sizeNormal colorDefault"
>
  header1
</span>
`;
